zerver/lib/rate_limiter.py

```diff
@@ -196,7 +261,7 @@ def incr_ratelimit(user, domain='all'):
                 pipe.watch(list_key)
 
                 # Get the last elem that we'll trim (so we can remove it from our sorted set)
-                last_val = pipe.lindex(list_key, max_api_calls(user) - 1)
+                last_val = pipe.lindex(list_key, max_api_calls(entity) - 1)
```

Comment:
This comes out fine in the end, but in an intermediate commit it has a type issue:
```
+                last_val = pipe.lindex(list_key, max_api_calls(entity.user) - 1)  # type: ignore # Just for this commit.
```

The same thing happens in a few other places, and in total they make the history a bit messy. In general we prefer to keep things clean and free of things like `type: ignore` throughout the history, to make it easier when going back and studying old commits to see how and why something became the way it is. Would you fix those?

The way I'd do that is to go into `git rebase -i` and move the `max_api_calls` and`max_api_window` commits before the commits that have these hacks. Then edit the hack commits to remove the hacks.